File: life-os-backend/src/main/java/com/lifeos/model/Task.java
package com.lifeos.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "tasks")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TaskStatus status;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TaskPriority priority;

    @Column(nullable = false)
    private boolean isRecurringBlock;

    private LocalDateTime scheduledStart;
    private LocalDateTime scheduledEnd;

    private Integer estimatedMinutes;

    @Column(name = "tags")
    private String tags; // Stored as comma-separated values

    public enum TaskStatus { PENDING, IN_PROGRESS, COMPLETED, MISSED }
    public enum TaskPriority { LOW, MEDIUM, HIGH }
}

File: life-os-ui/src/app/models/task.ts
export enum TaskStatus {
  PENDING = 'PENDING',
  IN_PROGRESS = 'IN_PROGRESS',
  COMPLETED = 'COMPLETED',
  MISSED = 'MISSED'
}

export enum TaskPriority {
  LOW = 'LOW',
  MEDIUM = 'MEDIUM',
  HIGH = 'HIGH'
}

export interface TaskRequest {
  title: string;
  description?: string;
  status?: TaskStatus;
  priority?: TaskPriority;
  isRecurringBlock: boolean;
  scheduledStart?: string;
  scheduledEnd?: string;
  estimatedMinutes?: number;
  tags?: string;
}

export interface TaskResponse {
  id: string;
  title: string;
  description: string;
  status: TaskStatus;
  priority: TaskPriority;
  isRecurringBlock: boolean;
  scheduledStart: string;
  scheduledEnd: string;
  estimatedMinutes: number;
  tags: string;
}
